src/utils/hoc.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import { observer } from 'mobx-react'
+import _ from 'lodash/fp'
+import StripedLoader from './StripedLoader'
+
+export let withNode = Component => props => {
+  let { tree, node, path } = props
+  node = node || (tree && path && tree.getNode(path))
+
+  if (!node) throw Error(`Node not provided, and couldn't find node at ${path}`)
+
+  return <Component {...props} node={node} />
+}
+
+export let withLoader = Component =>
+  observer(({ Loader = StripedLoader, node, ...props }) => (
+    <Loader loading={node && node.updating}>
+      <Component node={node} {...props} />
+    </Loader>
+  ))
+
+// I am a band-aid, please rip me off as quickly as possible
+export let withInlineLoader = Component =>
```

Comment:
I'd say that it makes more sense to have a style prop be applied to the loader itself instead of the wrapped component. Then you could get rid of this hoc.

Reply:
We did this intentionally to simplify the API of the HOC. The next change is going to remove this by allowing the loader to be overriden for the one component that uses it